File: App/lib/companies/company.dart
class Company {
  String? id;
  String name;
  String cnpj;
  String email;
  String description;

  Company(
      {this.id,
      required this.name,
      required this.cnpj,
      required this.email,
      required this.description});

  static Company fromMap(Map<String, dynamic> map) {
    return Company(
        id: map['id'],
        name: map['name'],
        cnpj: map['cnpj'],
        email: map['email'],
        description: map['description']);
  }
}

File: Api/Api/Contracts/Common/Company.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Api.Contracts.Common
{
    public class Company
    {
        [JsonRequired]
        [JsonPropertyName("name")]
        public string Name { get; set; }

        [JsonRequired]
        [JsonPropertyName("cnpj")]
        public string Cnpj { get; set; }

        [JsonRequired, EmailAddress]
        [JsonPropertyName("email")]
        public string Email { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        [JsonPropertyName("password")]
        public string Password { get; set; }

        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        [JsonPropertyName("description")]
        public string Description { get; set; }
    }
}